wgengine/monitor/monitor_windows.go

```diff
@@ -87,6 +87,7 @@ func (m *winMon) Close() error {
 	m.cancel()
 	m.pollTicker.Stop()
 	windows.SetEvent(m.closeHandle) // wakes up any reader blocked in Receive
+	windows.CloseHandle(m.closeHandle)
```

Comment:
Do we need to wait for the goroutine blocked in `WaitMultipleObjects` (or not-yet-blocked in it) to return before closing the handle?

Reply:
We do indeed. I did not consider scenario where m.closeHandle can be used after it was closed. Fixed now.

I did not want to introduce extra complexity. But I don't see other way.